From 93484e7e8e84bc64cdc24acf21ea395c233c64e2 Mon Sep 17 00:00:00 2001
From: Ziyuan Xu <xzy.xu@rock-chips.com>
Date: Mon, 29 Jul 2019 00:39:08 +0000
Subject: [PATCH 07/15] Client: adds nullptr checks whenever we use the
 fromWlSurface()

This is similar to commit 09f1d74b6278("Client: Add safer fromObject
function to scanner").

Signed-off-by: Ziyuan Xu <xzy.xu@rock-chips.com>
---
 src/client/qwaylandinputdevice.cpp | 16 ++++++++++------
 1 file changed, 10 insertions(+), 6 deletions(-)

diff --git a/src/client/qwaylandinputdevice.cpp b/src/client/qwaylandinputdevice.cpp
index b69789e..14343b8 100644
--- a/src/client/qwaylandinputdevice.cpp
+++ b/src/client/qwaylandinputdevice.cpp
@@ -678,8 +678,9 @@ void QWaylandInputDevice::Keyboard::keyboard_enter(uint32_t time, struct wl_surf
     if (!surface)
         return;
 
-
-    QWaylandWindow *window = QWaylandWindow::fromWlSurface(surface);
+    auto *window = QWaylandWindow::fromWlSurface(surface);
+    if (!window)
+        return; // Ignore foreign surfaces
     mFocus = window;
 
     mParent->mQDisplay->handleKeyboardFocusChanged(mParent);
@@ -690,10 +691,13 @@ void QWaylandInputDevice::Keyboard::keyboard_leave(uint32_t time, struct wl_surf
     Q_UNUSED(time);
     Q_UNUSED(surface);
 
-    if (surface) {
-        QWaylandWindow *window = QWaylandWindow::fromWlSurface(surface);
-        window->unfocus();
-    }
+    if (!surface)
+        return;
+
+    auto *window = QWaylandWindow::fromWlSurface(surface);
+    if (!window)
+        return; // Ignore foreign surfaces
+    window->unfocus();
 
     mFocus = nullptr;
 
-- 
2.20.1

